whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -624,39 +635,35 @@ private Domain createDomain(final RpslObject rpslObject, final String requestUrl
         domain.getEntitySearchResults().addAll(createContactEntities(rpslObject, requestUrl));
         return domain;
     }
+    private static VCard createContactVCard(final RpslObject rpslObject) {
+        return commonVcard(rpslObject).build();
+
+    }
 
     private static VCard createVCard(final RpslObject rpslObject) {
+        return commonVcard(rpslObject).addEmail(rpslObject.getValuesForAttribute(E_MAIL)).build();
+    }
+
+    private static VCardBuilder commonVcard(RpslObject rpslObject) {
         final VCardBuilder builder = new VCardBuilder();
         builder.addVersion();
 
         switch (rpslObject.getType()) {
-            case PERSON:
-                builder.addFn(rpslObject.getValueForAttribute(PERSON)).addKind(INDIVIDUAL);
-                break;
-            case MNTNER:
-                builder.addFn(rpslObject.getValueForAttribute(AttributeType.MNTNER)).addKind(INDIVIDUAL);
-                break;
-            case ORGANISATION:
-                builder.addFn(rpslObject.getValueForAttribute(ORG_NAME)).addKind(ORGANISATION);
-                break;
-            case ROLE:
-                builder.addFn(rpslObject.getValueForAttribute(ROLE)).addKind(GROUP);
-                break;
-            case IRT:
-                builder.addFn(rpslObject.getValueForAttribute(IRT)).addKind(GROUP);
-                break;
-            default:
-                break;
+            case PERSON -> builder.addFn(rpslObject.getValueForAttribute(PERSON)).addKind(INDIVIDUAL);
+            case MNTNER -> builder.addFn(rpslObject.getValueForAttribute(AttributeType.MNTNER)).addKind(INDIVIDUAL);
+            case ORGANISATION -> builder.addFn(rpslObject.getValueForAttribute(ORG_NAME)).addKind(ORGANISATION);
+            case ROLE -> builder.addFn(rpslObject.getValueForAttribute(ROLE)).addKind(GROUP);
+            case IRT -> builder.addFn(rpslObject.getValueForAttribute(IRT)).addKind(GROUP);
+            default -> {
+            }
         }
-        builder.addAdr(rpslObject.getValuesForAttribute(ADDRESS))
-                .addTel(rpslObject.getValuesForAttribute(PHONE))
+        builder.addTel(rpslObject.getValuesForAttribute(PHONE))
                 .addFax(rpslObject.getValuesForAttribute(FAX_NO))
-                .addEmail(rpslObject.getValuesForAttribute(E_MAIL))
+                .addAdr(rpslObject.getValuesForAttribute(ADDRESS))
```

Comment:
Can you remove email in a separate PR? It's not related to the redacted feature

Reply:
as far as I understand we need to specify all of attributes that we are not specifying in the output because of some internal reason. This is why I am taking in account the email here and trying to fix the email as well. Let me know if I am wrong.

Reply:
Done